flask_oasschema.py

```diff
@@ -63,12 +88,31 @@ def foo(param):
             ...
     """
     def wrapper(fn):
+        def convert_type(string_value):
+            str_value = string_value.decode('utf8')
+            if str_value.isnumeric():
+                return int(string_value)
+            else:
+                return str_value
+
         @wraps(fn)
         def decorated(*args, **kwargs):
             uri_path = request.url_rule.rule.replace("<", "{").replace(">", "}")
             method = request.method.lower()
             schema = current_app.extensions['oas_schema']
-            validate(request.get_json(), extract_schema(schema, uri_path, method))
+
+            try:
+                validate(request.get_json(), extract_body_schema(schema, uri_path, method))
+            except ValidationError:
```

Comment:
changed it to consider the query string instead of the body only in the case of GET